src/dmd/semantic3.d

```diff
@@ -241,6 +241,20 @@ private extern(C++) final class Semantic3Visitor : Visitor
         {
             if (funcdecl.storage_class & STC.inference)
             {
+                /* https://issues.dlang.org/show_bug.cgi?id=18672
+                   If the parent struct has a destructor that is
+                   `@safe`, then the `opAssign` function is `@trusted`
+                 */
+                auto sd = funcdecl.parent.isStructDeclaration();
```

Comment:
Shouldn't it check for classes as well?

Reply:
identity opAssign is not allowed for classes: https://dlang.org/spec/operatoroverloading.html

Reply:
`opAssign` is allowed for classes if the parameter is of a different type than the class. I tried this and got the same error as for the struct example.